source/slang/check.cpp

```diff
@@ -6601,11 +6601,15 @@ namespace Slang
             candidate.subst = genSubst;
             auto& checkedArgs = genSubst->args;
 
-            int aa = 0;
+            uint32_t aa = 0;
             for (auto memberRef : getMembers(genericDeclRef))
             {
                 if (auto typeParamRef = memberRef.As<GenericTypeParamDecl>())
                 {
+                    if (aa >= context.argCount)
+                    {
+                        return false;
```

Comment:
Should we be producing a diagnostic in this case (conditionally, similar to how the other overload-resolving functions only emit errors once it is known that this is the "right" overload)?

Reply:
I am not sure how to tell when "it is known that this is the right overload", since the purpose of this function seemed to be to test whether a candidate is applicable,  and it does not possess the information on whether this is the only candidate...